lib/Dialect/FIRRTL/Transforms/ExtractInstances.cpp

```diff
@@ -568,12 +587,36 @@ void ExtractInstancesPass::extractInstances() {
           if (!innerRef)
             continue;
           if (innerRef.getModule() == parent.moduleNameAttr() &&
-              innerRef.getName() == newInst.inner_symAttr())
+              innerRef.getName() == inst.inner_symAttr())
             break;
         }
-        assert(nlaIdx < nlaLen && "instance not found in its own NLA");
+        if (nlaIdx >= nlaLen) {
+          // This usually happens if the instance is extracted into multiple
+          // parents (because the current parent module is multiply
+          // instantiated). In that case NLAs that were specific to one instance
+          // may have been moved when we arrive at the second instance, and the
+          // NLA is already updated.
+          LLVM_DEBUG(llvm::dbgs() << "    - Instance no longer in path\n");
+          return true;
+        }
```

Comment:
I had trouble glancing at the if condition and knowing what it meant, maybe a comment before "Handle the case if the NLA does not contain the instance op"

Reply:
:+1: Added a comment